frontend/automation/cypress/e2e/wallet-connection.cy.js

```diff
@@ -0,0 +1,63 @@
+const { unlockWallet } = require('../support/helpers')
+
+const testIds = {
+  SELECTION_MODAL: 'wallet-selection-modal',
+  REJECT_CONNECTION_BUTTON: 'wallet-connection-reject',
+  APPROVE_SELECTION_BUTTON: 'wallet-selection-button'
+}
+
+describe('wallet connection', () => {
+  let walletName
+  let passphrase
+
+  before(() => {
+    cy.clean()
+    cy.backend().then(handler => {
+      cy.setVegaHome(handler)
+      cy.restoreNetwork(handler)
+      cy.restoreWallet(handler)
+    })
+  })
+
+  beforeEach(() => {
+    passphrase = Cypress.env('testWalletPassphrase')
+    walletName = Cypress.env('testWalletName')
+    cy.waitForHome()
+    unlockWallet(walletName, passphrase)
+  })
+
+  it('handles approval', () => {
+    const MOCK_HOSTNAME = 'https://best-blockchain.app'
+    cy.sendConnectionRequest(MOCK_HOSTNAME)
+
+    cy.getByTestId(testIds.SELECTION_MODAL).should('exist')
+    cy.getByTestId(testIds.SELECTION_MODAL).should('be.visible')
+    cy.getByTestId(testIds.APPROVE_SELECTION_BUTTON).click()
+
+    cy.getByTestId('input-passphrase').type(passphrase)
+    cy.getByTestId('input-submit').click()
+
+    cy.getByTestId('toast').should(
+      'have.text',
+      'The connection to the wallet has been successfully established.'
+    )
+  })
+
+  // it('handles rejection', () => {
+  // @TODO: for some mysterious reason this is failing
```

Comment:
created this, since we've spent way too much time already debugging cypress
https://github.com/vegaprotocol/vegawallet-desktop/issues/355